### PR TITLE
Added console command abstract and json optional for transformer

### PR DESCRIPTION
**Pull request by stephenwf**
This is a backwards compatible addition to the queue transformer for non-gearman transforms where serialisation is not needed.

(Sorry last one was from wrong branch)

### CODE REVIEW

**Comment by stephenwf** [comment]
Should I add `symfony/console` and `ext-newrelic` into the suggest field of `composer.json`?

File: src/Queue/BasicTransformer.php [hunk 1]
```diff
@@ -49,11 +49,14 @@ public function getSdk(QueueItem $item)
         }
     }
 
-    public function transform(QueueItem $item)
+    public function transform(QueueItem $item, bool $asJson = true)
```
Comment (giorgiosironi) [inline src/Queue/BasicTransformer.php]: can you document the possible return types of this method with `@return`?

File: src/Queue/QueueItemTransformer.php [hunk 2]
```diff
@@ -4,5 +4,5 @@
 
 interface QueueItemTransformer
 {
-    public function transform(QueueItem $item);
```
Comment (giorgiosironi) [inline src/Queue/QueueItemTransformer.php]: actually better to document it here, in the interface

File: src/Command/QueueCommand.php [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+abstract class QueueCommand extends Command
+{
+    protected $logger;
+    protected $queue;
+    protected $transformer;
+    protected $limit;
+    protected $monitoring;
+    protected $topic;
+
+    public function __construct(string $topic, LoggerInterface $logger, WatchableQueue $queue, QueueItemTransformer $transformer, Monitoring $monitoring, callable $limit)
```
Comment (giorgiosironi) [inline src/Command/QueueCommand.php]: should not be a `$topic` but a `$queue`. But actually we already pass in `$queue` so if the only thing this is doing with `string $topic` is to log its its name, I would remove it. If we really need it for logging we can add `__toString()` to `WatchableQueue`

**Comment by giorgiosironi** [comment]
We should add `symfony/console` (maybe even as a requirement rather than as a suggestion), while `ext/newrelic` is in the suggestions of dependency `logging-sdk-php`. We cannot put `ext/newrelic` in strict requirements as it's usually missing locally.

**Comment by stephenwf** [comment]
Sounds good, I've added console as a requirement for now. I think chances are if a project is using this library, especially the queue, they will have a symfony console component set up to run it.

File: src/Command/QueueCommand.php (hunk 3 above)
Comment (stephenwf) [inline src/Command/QueueCommand.php]: I like the toString idea, that should work nicely

**Comment by stephenwf** [comment]
@giorgiosironi Added those changes now

File: src/Command/QueueCommand.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    abstract protected function process(InputInterface $input, QueueItem $item);
+
+    protected function configure()
+    {
+        $this
+            ->setName('queue:watch')
+            ->setDescription('Watches SQS for changes to articles, ')
+            ->addOption('drop', 'd', InputOption::VALUE_NONE);
```
Comment (giorgiosironi) [inline src/Command/QueueCommand.php]: this `drop` option does not seem used?

**Comment by giorgiosironi** [comment]
Approved but fix the build first, possibly just some coding standards.

**Pull request merged**